pennylane/interfaces/tensorflow.py

```diff
@@ -111,19 +111,20 @@ def _jac_restructured(jacs, tapes):
     """
     start = 0
     jacs_nested = []
-    for i, tape in enumerate(tapes):
+    for tape in tapes:
         num_meas = len(tape.measurements)
         num_params = len(tape.trainable_params)
 
         tape_jacs = tuple(jacs[start : start + num_meas * num_params])
         tape_jacs = tuple(
-            tuple(tape_jacs[i * num_params : (i + 1) * num_params]) for i in range(num_meas)
+            tuple(tape_jacs[j * num_params : (j + 1) * num_params]) for j in range(num_meas)
         )
 
         while isinstance(tape_jacs, tuple) and len(tape_jacs) == 1:
             tape_jacs = tape_jacs[0]
 
         jacs_nested.append(tape_jacs)
+        start += num_meas * num_params

```

Comment:
Was `start` not changed previously? :thinking:

Reply:
It was not. Previously the test file only calls `qml.execute` on single tapes, so this was a bug that slipped through. Batching requires calling `qml.execute` with multiple tapes now.